### PR TITLE
[PROTOCOL-720] Create Loan With NFT V2

### PR DESCRIPTION
**Pull request by passabilities**
* Fixes NFT loan interest (loans with NFTs should have 0% interest)
* Splits existing logic for creating a loan with NFT V1 & consensus types into separate functions
* Moves `LibCreateLoan` into separate file
* Adds functionality to take out a loan with NFT V2
* Uses the same function to take out loans with NFT V1 and V2 by passing in data to decode

### CODE REVIEW

**Review by mkhoury93 — DISMISSED** [review]

**Review by admazzola — APPROVED** [review]

**Review by mkhoury93 — APPROVED** [review]

**Review by dumebi — APPROVED** [review]

**Comment by passabilities** [comment]
Please review #448 first!

File: deploy/protocol.ts [hunk 1]
```diff
@@ -69,29 +69,29 @@ const deployProtocol: DeployFunction = async (hre) => {
...
     },
     {
       contract: 'AssetSettingsFacet',
-      skipIfAlreadyDeployed: true,
+      skipIfAlreadyDeployed: false,
     },
     {
       contract: 'PausableFacet',
-      skipIfAlreadyDeployed: true,
+      skipIfAlreadyDeployed: false,
```
Comment (sypeer) [inline deploy/protocol.ts:94]: why are we changing this here again?

**Review by sypeer — DISMISSED** [review]
Check the failing test please (not sure why the aave test is failing on this)

**Review by dumebi — APPROVED** [review]

**Review by sypeer — APPROVED** [review]

File: contracts/market/libraries/LibLoans.sol [hunk 2]
```diff
@@ -201,13 +201,16 @@ library LibLoans {
         view
         returns (uint16 ratio_)
     {
-        ratio_ = uint16(
-            (uint64(loan(loanID).duration) * loan(loanID).interestRate) /
-                SECONDS_PER_YEAR
-        );
+        if (loan(loanID).interestRate > 0) {
+            ratio_ = uint16(
+                (uint64(loan(loanID).duration) * loan(loanID).interestRate) /
```
Comment (admazzola) [inline contracts/market/libraries/LibLoans.sol:206]: we should double check this casting will not cause any issues

**Review by admazzola — APPROVED** [review]

**Review by dumebi — APPROVED** [review]

**Review by mkhoury93 — APPROVED** [review]

**Review by sypeer — APPROVED** [review]

**Pull request merged**